frontend/src/Frontend/UI/Widgets.hs

```diff
@@ -889,92 +886,49 @@ uiGasPriceInputField eReset conf = dimensionalInputFeedbackWrapper (Just "KDA")
 
 -- | Let the user pick a chain id.
 userChainIdSelect
-  :: ( MonadWidget t m
-     , HasNetwork model t
-     )
-  => model
-  -> m ( Dropdown t (Maybe ChainId) )
-userChainIdSelect m =
-  userChainIdSelectWithPreselect m True (constDyn Nothing)
-
--- | Let the user pick a chain id but preselect a value
-userChainIdSelectWithPreselect
-  :: (MonadWidget t m, HasNetwork model t
-     )
-  => model
-  -> Bool
-  -> Dynamic t (Maybe ChainId)
-  -> m ( Dropdown t (Maybe ChainId) )
-userChainIdSelectWithPreselect m inlineLabel mChainId =
-  mkLabeledClsInput inlineLabel "Chain ID" (uiChainSelection mNodeInfo mChainId)
-  where
-    mNodeInfo = (^? to rights . _head) <$> m ^. network_selectedNodes
-
-uiChainSelection
   :: MonadWidget t m
-  => Dynamic t (Maybe NodeInfo)
-  -> Dynamic t (Maybe ChainId)
-  -> CssClass
+  => Dynamic t [ChainId]
   -> m ( Dropdown t (Maybe ChainId) )
-uiChainSelection info mPreselected cls = do
-  onPreselected <- tagOnPostBuild mPreselected
-  let
-    chains = map (id &&& _chainId) . maybe [] getChains <$> info
-    mkPlaceHolder cChains = if null cChains then "No chains available" else "Select chain"
-    mkOptions cs = Map.fromList $ (Nothing, mkPlaceHolder cs) : map (first Just) cs
-
-    staticCls = cls <> "select"
-    mkDynCls v = if isNothing v then "select_mandatory_missing" else mempty
-
-  rec
-    let allCls = renderClass <$> fmap mkDynCls (_dropdown_value ddE) <> pure staticCls
-        cfg = def
-          & dropdownConfig_attributes .~ (("class" =:) <$> allCls)
-          & dropdownConfig_setValue .~ onPreselected
-
-    ddE <- dropdown Nothing (mkOptions <$> chains) cfg
-  pure ddE
+userChainIdSelect options = do
+  mkLabeledClsInput True "Chain ID" (uiChainSelectionWithUpdate options never)
 
 uiChainSelectionWithUpdate
   :: MonadWidget t m
-  => Dynamic t (Maybe NodeInfo)
+  => Dynamic t [ChainId]
   -> Event t (Maybe ChainId)
   -> CssClass
   -> m ( Dropdown t (Maybe ChainId) )
-uiChainSelectionWithUpdate info onPreselected cls = do
+uiChainSelectionWithUpdate options onPreselected cls = do
   let
-    chains = map (id &&& _chainId) . maybe [] getChains <$> info
+    chains = map (id &&& _chainId) <$> options
     mkPlaceHolder cChains = if null cChains then "No chains available" else "Select chain"
     mkOptions cs = Map.fromList $ (Nothing, mkPlaceHolder cs) : map (first Just) cs
 
     staticCls = cls <> "select"
     mkDynCls v = if isNothing v then "select_mandatory_missing" else mempty
+  pb <- getPostBuild
+  let optionsEv = leftmost [current options <@ pb, updated options]
```

Comment:
Is this `tagOnPostBuild` ? It uses the `updated` event too, I think

Reply:
Oh, I hadn't realized `tagOnPostBuild` tagged _beyond_ on postBuild. But I see it uses `tagPromptlyDyn` so I'd rather steer clear of that.